src/hpa.c

```diff
@@ -500,7 +587,7 @@ hpa_purge_start_hp(hpa_purge_batch_t *b, psset_t *psset) {
 static inline void
 hpa_purge_finish_hp(
     tsdn_t *tsdn, hpa_shard_t *shard, hpa_purge_item_t *hp_item) {
-	if (hp_item->dehugify) {
+	if (hp_item->dehugify && hpa_is_hugify_lazy(shard)) {
 		shard->stats.ndehugifies++;
 	}
```

Comment:
I am getting confused about the option naming especially when dehugify is guarded by a hugify option.  It feels that policies about when to promote an demote a huge page are distinct.  Likewise, for actions taken when a page needs to be purged.  Furthermore, there is going to be a cross-product effect since some of those options will want to be combined.